packages/@lwc/engine-core/src/framework/template.ts

```diff
@@ -99,26 +100,30 @@ function validateSlots(vm: VM) {
     }
 }
 
-function validateLightDomTemplate(template: Template, vm: VM) {
-    assertNotProd(); // should never leak to prod mode
+function checkHasMatchingRenderMode(template: Template, vm: VM) {
+    // don't validate in prod environments where reporting is disabled
+    if (process.env.NODE_ENV === 'production' && !isReportingEnabled()) {
+        return;
+    }
+    // don't validate the default empty template - it is not inherently light or shadow
     if (template === defaultEmptyTemplate) {
         return;
     }
-    if (vm.renderMode === RenderMode.Light) {
-        if (template.renderMode !== 'light') {
-            logError(
-                `Light DOM components can't render shadow DOM templates. Add an 'lwc:render-mode="light"' directive to the root template tag of ${getComponentTag(
-                    vm
-                )}.`
-            );
-        }
-    } else {
-        if (!isUndefined(template.renderMode)) {
-            logError(
-                `Shadow DOM components template can't render light DOM templates. Either remove the 'lwc:render-mode' directive from ${getComponentTag(
-                    vm
-                )} or set it to 'lwc:render-mode="shadow"`
-            );
+    // TODO [#4663]: `renderMode` mismatch between template and component causes `console.error` but no error
```

Comment:
```suggestion
    // TODO [#4663]: `renderMode` mismatch between template and component causes `console.error` but no error
    // Note that `undefined` means shadow in this case, because shadow is the default.
```